src/unix.rs

```diff
@@ -130,6 +132,18 @@ impl MmapInner {
         }
     }
 
+    pub fn set_protection(&mut self, prot: Protection) -> io::Result<()> {
+        let result = unsafe { libc::mprotect(self.ptr,
```

Comment:
I think this is going to have the same issue with page offset with a file-backed mmap. It would be great to have a test that creates a file mmap with offset and changes protection from read+write to read, even if all it's doing is making sure that the system calls don't fail.